pepper-apis/src/main/resources/db-changes/schema/DDP-6983-activity-instance-creation-mutex-table.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="mocana" id="20210919-activity_instance_creation_mutex_table">
+        <createTable tableName="activity_instance_creation_mutex">
+            <column name="activity_instance_creation_mutex_id" type="bigint" autoIncrement="true" startWith="1">
+                <constraints primaryKey="true" primaryKeyName="activity_instance_creation_mutex_pk"/>
+            </column>
+            <column name="participant_id" type="bigint">
+                <constraints nullable="false"
+                             references="user(user_id)"
+                             foreignKeyName="act_instance_mutex_participant_fk"/>
+            </column>
+            <column name="study_activity_id" type="bigint">
+                <constraints nullable="true"
```

Comment:
Good question! I started with a table design with a composite primary key of study_activity_id and participant_id. So study_activity_id was NOT nullable.
But that version of my table did not pass the unit tests.
The test POSTs a request to create a activity instance for a study activity guid that does not exist.
So code had to handle bogus study activities.
So had two choices:
1) Add a query for the study activity to see if it exists and either return an error if bogus so as to avoid inserting a NULL in the mutex table.
2) Allow the null to be stored in the mutex table.
I did not want to add new queries (we do too many queries already!), so made the column nullable. The next query that checks the validation will produce the error.
And everything works fine/there is no harm with nullable column.
So that is the story.